### ISSUE TITLE
demo2d is broken

### ISSUE THREAD

**Issue by touretzkyds** [touretzkyds/KnowledgeGraphDemo#37]
The latest version of demo2d was working fine for me earlier today. I just tried it again and it has stopped working, although the code has not changed. This must be a change to the Devel graph. Here is the error message.

![broken-demo](https://user-images.githubusercontent.com/1978092/175204663-ede03e9c-bb3b-4d31-95aa-fd61b3dce04b.png)

**Comment by nhadlaw** [comment]
What action(s) led to this issue or did it just occur on loading?

**Comment by touretzkyds** [comment]
The error occurs on loading. I'm loading from http://www.cs.cmu.edu/~dst/test/KnowledgeGraphDemo/demo2d.html

**Comment by deyaliao** [comment]
Hi, it should work now. When I reran the scraping algorithm, I ended up with one blank entry (the operation timed out, which it often does randomly for different taxa), which broke the demo.

**Comment by touretzkyds** [comment]
Yep. It's beautiful again. Looks like the scraping code needs some safety checks installed so if we use it again (or a later team member uses it as a starting point), it will be smarter about timeouts. Can you add that feature?

**Comment by deyaliao** [comment]
What exactly do you mean about safety checks? Initially I would let the operation time out, but then that was a hassle when I had to rereun the algorithm simply because of 1-2 timeouts. Now, I just let the scraping algorithm continue; perhaps, I could make note of which ones time out at the end so I don't make the same mistake as I do with this issue again? Is that what you mean?

**Comment by touretzkyds** [comment]
I was being vague because I haven't actually looked at your code or tried to run it. What I had in mind was something to alert the user that the data file is incomplete because some retrievals timed out. If it's possible to point out exactly which retrievals failed, perhaps the user could just run those few items again. But I'm just guessing here. The important thing is that the user be warned that the data file is incomplete and must not be used without repairing it first. Hopefully you'll be able to tell them exactly what needs to be repaired.
